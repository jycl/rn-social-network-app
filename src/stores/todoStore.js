import { observable, computed, toJS, action } from "mobx";
import { getTodosForUser } from "../services/APIService";

/**
 * TodoStore is an MobX store that manages the state values of components related
 * to the todo list to display.
 *
 * State params:
 * @param {Array} rawTodoList Array of todos for user retrieved from backend API
 * @param {Array} todoList list of todos that has been converted from MobX object
 */
class TodoStore {
  @observable
  rawTodoList = [];

  /**
   * Retrieve user list from backend and save to store
   */
  @action
  loadTodoList = userId => {
    getTodosForUser(userId).then(todos => {
      this.rawTodoList = todos;
    });
  };

  @computed
  get todoList() {
    return toJS(this.rawTodoList);
  }
}
const todoStore = new TodoStore();
export default todoStore;
