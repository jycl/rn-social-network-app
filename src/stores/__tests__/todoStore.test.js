import todoStore from "../todoStore";

describe("TodoStore tests ", () => {
  it("Snapshot test of TodoStore", () => {
    expect(todoStore).toMatchSnapshot();
  });

  it("Check initial state values and changes", () => {
    expect(todoStore.rawTodoList).toHaveLength(0);
    expect(todoStore.todoList).toHaveLength(0);

    const testTodoList = [
      {
        id: 1,
        title: "Jogging",
        completed: true
      },
      {
        id: 2,
        title: "Laundry",
        completed: false
      }
    ];
    todoStore.loadTodoList = jest.fn().mockImplementation(() => {
      todoStore.rawTodoList = testTodoList;
    });

    todoStore.loadTodoList("1");
    expect(todoStore.todoList).toHaveLength(2);
    expect(todoStore.todoList[0].completed).toBe(true);

    todoStore.clearData();
    expect(todoStore.todoList).toHaveLength(0);
  });
});
