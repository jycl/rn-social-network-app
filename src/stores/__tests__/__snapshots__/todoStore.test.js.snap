// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoStore tests  Snapshot test of TodoStore 1`] = `TodoStore {}`;
